Core/GDCore/Project/Object.cpp

```diff
@@ -20,11 +20,20 @@ namespace gd {
 
 Object::~Object() {}
 
-Object::Object(const gd::String& name_, const gd::String& type_, std::unique_ptr<gd::ObjectConfiguration> configuration_)
+Object::Object(const gd::String& name_,
+               const gd::String& type_,
+               std::unique_ptr<gd::ObjectConfiguration> configuration_)
     : name(name_), configuration(std::move(configuration_)) {
       SetType(type_);
     }
 
+Object::Object(const gd::String& name_,
+               const gd::String& type_,
+               gd::ObjectConfiguration& configuration_)

```

Comment:
I think it's super risky to create a configuration from a point to a reference, because it means the unique_ptr will take ownership of the lifetime of the reference. But there is a risk someone would use this by passing a ObjectConfiguration created on the stack:

```
{
  gd::ObjectConfiguration configuration; // Unlikely, but as you can see, easy to write
  gd::Object object("name", "sometype", configuration);
}
// Boom, configuration was released because out of scope and the unique_ptr is pointing at something deleted.
```

Instead, take a pointer (`gd::ObjectConfiguration*`) to show that this is a raw pointer, so the reader must check who is taking ownership of it (answer being in the documentation generally).